src/EFCore.PG.NTS/Storage/Internal/NpgsqlGeometryTypeMapping.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Data.Common;
+using System.Text;
+using Microsoft.EntityFrameworkCore.Storage;
+using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
+using GeoAPI.Geometries;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+using NetTopologySuite.IO;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.Mapping;
+using NpgsqlTypes;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal
+{
+    public class NpgsqlGeometryTypeMapping<TGeometry> : RelationalGeometryTypeMapping<TGeometry, TGeometry>
+    {
+        readonly bool _isGeography;
+
+        public NpgsqlGeometryTypeMapping(string storeType) : base(new NullValueConverter(), storeType)
+            => _isGeography = IsGeography(storeType);
+
+        protected NpgsqlGeometryTypeMapping(RelationalTypeMappingParameters parameters)
+            : base(parameters) {}
+
+        protected override RelationalTypeMapping Clone(RelationalTypeMappingParameters parameters)
+            => new NpgsqlGeometryTypeMapping<TGeometry>(parameters);
+
+        protected override void ConfigureParameter([NotNull] DbParameter parameter)
+        {
+            base.ConfigureParameter(parameter);
+
+            ((NpgsqlParameter)parameter).NpgsqlDbType = _isGeography ? NpgsqlDbType.Geography : NpgsqlDbType.Geometry;
+        }
+
+        protected override string GenerateNonNullSqlLiteral(object value)
+        {
+            var geometry = (IGeometry)value;
+            var builder = new StringBuilder();
+
+            builder.Append(_isGeography ? "GEOGRAPHY " : "GEOMETRY '");
+
+            if (geometry.SRID > 0)
+            {
+                builder
+                    .Append("SRID=")
+                    .Append(geometry.SRID)
+                    .Append(";");
```

Comment:
nit: Could we just append the `char`?

Reply:
Changed although I don't think there's any value in it.